radix-engine/src/blueprints/resource/bucket_common.rs

```diff
@@ -0,0 +1,35 @@
+use crate::blueprints::resource::*;
+use crate::errors::RuntimeError;
+use crate::kernel::kernel_api::{KernelNodeApi, KernelSubstateApi};
+use crate::types::*;
+use radix_engine_interface::api::substate_api::LockFlags;
+use radix_engine_interface::api::ClientSubstateApi;
+use radix_engine_interface::blueprints::resource::*;
+
+#[derive(Debug, Clone, PartialEq, Eq, ScryptoSbor)]
+pub enum BucketError {
+    ResourceError(ResourceError),
+    ProofError(ProofError),
+    MismatchingResource,
+    InvalidAmount,
+}
+
+#[derive(Debug, Clone, PartialEq, Eq, ScryptoSbor)]
+pub struct BucketInfoSubstate {
+    pub resource_address: ResourceAddress,
```

Comment:
I think we can remove this since we can just get this resource address through the type_info api call. Or can we atleast add a TODO to fix this.

Reply:
I thought I added a TODO somewhere else. Will update if not